realm/realm-library/src/main/cpp/util.hpp

```diff
@@ -343,7 +343,7 @@ inline bool TypeValid(JNIEnv* env, T* pTable, jlong columnIndex, int expectColTy
     int colType = pTable->get_column_type(col);
     if (colType != expectColType) {
         realm::jni_util::Log::e("Expected columnType %1, but got %2.", expectColType, pTable->get_column_type(col));
-        ThrowException(env, IllegalArgument, "ColumnType invalid.");
+        ThrowException(env, IllegalArgument, "ColumnType of '" + ((std::string) pTable->get_column_name(col)) + "' is invalid.");
```

Comment:
shall we use `std::string(pTable->get_column_name(col))` instead to avoid c-style casting?